src/cascadia/TerminalApp/PowershellCoreProfileGenerator.cpp

```diff
@@ -10,81 +10,320 @@
+                out.emplace_back(PowerShellInstance{ std::stoi(versionedPath.filename()),
+                                                     PowerShellFlags::Traditional | flags | previewFlag,
+                                                     executable });
+            }
+        }
+    }
 }
 
 // Function Description:
-// - Returns true if the user has installed PowerShell Core.
+// - Finds the store package, if one exists, for a given package family name
 // Arguments:
-// - A string that contains an environment-variable string in the form: %variableName%.
-// - A ref of a path that receives the result of PowerShell Core pwsh.exe full path.
+// - packageFamilyName: the package family name
 // Return Value:
-// - true iff powershell core (pwsh.exe) is present in the given path
-bool PowershellCoreProfileGenerator::_isPowerShellCoreInstalledInPath(const std::wstring_view programFileEnv, std::filesystem::path& cmdline)
+// - a package, or nullptr.
+static winrt::Windows::ApplicationModel::Package _getStorePackage(const std::wstring_view packageFamilyName) noexcept
+try
+{
+    winrt::Windows::Management::Deployment::PackageManager packageManager;
+    auto foundPackages = packageManager.FindPackagesForUser(L"", packageFamilyName);
+    auto iterator = foundPackages.First();
+    if (!iterator.HasCurrent())
+    {
+        return nullptr;
+    }
+    return iterator.Current();
+}
+catch (...)
+{
+    LOG_CAUGHT_EXCEPTION();
+    return nullptr;
+}
+
+// Function Description:
+// - Finds all powershell instances that have App Execution Aliases in the standard location
+// Arguments:
+// - out: the list into which to accumulate these instances.
+static void _accumulateStorePowerShellInstances(std::vector<PowerShellInstance>& out)
 {
-    std::wstring programFileEnvNulTerm{ programFileEnv };
-    std::filesystem::path psCorePath{ wil::ExpandEnvironmentStringsW<std::wstring>(programFileEnvNulTerm.data()) };
-    psCorePath /= L"PowerShell";
-    if (std::filesystem::exists(psCorePath))
+    wil::unique_cotaskmem_string localAppDataFolder;
+    if (FAILED(SHGetKnownFolderPath(FOLDERID_LocalAppData, 0, 0, &localAppDataFolder)))
     {
-        for (auto& p : std::filesystem::directory_iterator(psCorePath))
+        return;
+    }
+
+    std::filesystem::path appExecAliasPath{ localAppDataFolder.get() };
+    appExecAliasPath /= L"Microsoft";
+    appExecAliasPath /= L"WindowsApps";
+
+    if (std::filesystem::exists(appExecAliasPath))
+    {
+        // App execution aliases for preview powershell
+        const auto previewPath = appExecAliasPath / POWERSHELL_PREVIEW_PFN;
+        if (std::filesystem::exists(previewPath))
+        {
+            auto previewPackage = _getStorePackage(POWERSHELL_PREVIEW_PFN);
+            if (previewPackage)
+            {
+                out.emplace_back(PowerShellInstance{
+                    static_cast<int>(previewPackage.Id().Version().Major),

```

Comment:
If audit mode were on in this project, I think it would carp at you for static casting here.